core/arch/arm/kernel/thread.c

```diff
@@ -1141,7 +1156,9 @@ static uint32_t rpc_cmd_nolock(uint32_t cmd, size_t num_params,
 	const size_t params_size = sizeof(struct optee_msg_param) * num_params;
 	size_t n;
 
-	TEE_ASSERT(arg && carg && num_params <= RPC_MAX_NUM_PARAMS);
+	if (!arg || !carg || num_params > RPC_MAX_NUM_PARAMS)
```

Comment:
A an `assert()` should be enough here, this function is only called via `thread_rpc_cmd()` which in its turn only is called from wrapper functions.